### PR TITLE
add module_unload_list, module_load, envs to Resources;

### PR DESCRIPTION
**Pull request by felix5572**
add module_unload_list, module_load, envs to Resources; #45

### CODE REVIEW

File: dpdispatcher/machine.py [hunk 1]
```diff
@@ -153,14 +156,34 @@ def gen_script_custom_flags_lines(self, job):
...
+
+        module_unload_part = ""
+        module_unload_list = job.resources.module_unload_list
+        for ii in module_unload_list:
+            module_unload_part += f"module unload {ii}\n"
+
+        module_load_part = ""
+        module_list = job.resources.module_list
+        for ii in module_list:
+            module_load_part += f"module load {ii}\n"
```
Comment (njzjz) [inline dpdispatcher/machine.py:168]: I heard that, `+=` in Python is slower than `join`.

Comment (njzjz) [reply dpdispatcher/machine.py:168]: See https://stackoverflow.com/a/39312172

File: tests/jsons/machine_slurm.json [hunk 2]
```diff
@@ -0,0 +1,29 @@
+{
+    "machine":{
+        "batch_type": "Slurm",
+        "context_type": "SSHContext",
+        "local_root": "./test_context_dir",
+        "remote_root": "/home/fengbo/work_path_dpdispatcher_test",
+        "remote_profile": {
+            "hostname": "101.200.149.59",
+            "username": "fengbo"
```
Comment (AnguseZhang) [inline tests/jsons/machine_slurm.json:9]: We should not expose IP

**Review by AnguseZhang — APPROVED** [review]

File: tests/jsons/machine_slurm.json (hunk 2 above)
Comment (felix5572) [reply tests/jsons/machine_slurm.json]: fixed

**Pull request merged**